### PR TITLE
CBG-812 - Fixes infinite loop caused by stale view query when running compact

### PR DESCRIPTION
**Pull request by bbrks**

### CODE REVIEW

File: db/query.go [hunk 1]
```diff
@@ -512,11 +512,12 @@ func (context *DatabaseContext) QueryTombstones(olderThan time.Time, limit int,
 	// View Query
 	if context.Options.UseViews {
 		opts := Body{"stale": "ok"}
-		opts[QueryParamStartKey] = 1
-		opts[QueryParamEndKey] = olderThan.Unix()
 		if limit != 0 {
+			opts = Body{"stale": false}
```
Comment (bbrks) [inline db/query.go]: @adamcfraser Maybe I should evaluate the `consistencyMode` parameter instead of relying on limit?

Comment (adamcfraser) [reply db/query.go]: Yes I think you're right - we should just be using that, and it looks like that's already doing the right thing (always requestPlus).

Comment (bbrks) [reply db/query.go]: Updated to change the param to a stale bool that I can use for both Views and N1QL Query.

It looks like there were unused params in the N1QL query that were probably copy/paste errors. Give it a look and see if you disagree.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/query.go [hunk 2]
```diff
@@ -507,13 +507,18 @@ func (context *DatabaseContext) QueryAllDocs(startKey string, endKey string) (sg
 	return context.N1QLQueryWithStats(QueryTypeAllDocs, allDocsQueryStatement, params, gocb.RequestPlus, QueryAllDocs.adhoc)
 }
 
-func (context *DatabaseContext) QueryTombstones(olderThan time.Time, limit int, consistencyMode gocb.ConsistencyMode) (sgbucket.QueryResultIterator, error) {
+func (context *DatabaseContext) QueryTombstones(olderThan time.Time, limit int, stale bool) (sgbucket.QueryResultIterator, error) {
```
[the remark this answers is not in the document]
Comment (bbrks) [reply db/query.go]: Sure, makes sense.

**Review by adamcfraser — APPROVED** [review]
LGTM

**Pull request merged**